tests/test_cases/test_discovery/test_discovery.py

```diff
@@ -134,8 +134,8 @@ def access_single_bit_erroneous(dut):
     dut.stream_in_data[bit] <= 1
     yield Timer(10)
 
-@cocotb.test(expect_error=cocotb.SIM_NAME in ["Icarus Verilog"],
-             expect_fail=cocotb.SIM_NAME in ["Riviera-PRO"])
+@cocotb.test(expect_error=cocotb.SIM_NAME.lower().startswith(("Icarus Verilog")),
```

Comment:
Yes you are correct, and the second part of the name can be dropped as well.  I will update.